src/stcal/saturation/saturation.py

```diff
@@ -102,6 +101,25 @@ def flag_saturated_pixels(
             del flagarray
             del flaglowarray
 
+            if read_pattern is not None:
+                if group == 2:
+                    # Identify groups which we wouldn't expect to saturate by the third group,
+                    # on the basis of the first group
+                    mask = data[ints, 0, ...] / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
+
+                    # Identify groups with suspiciously large values in the second group
+                    mask &= data[ints, 1, ...] > sat_thresh / len(read_pattern[1])
+
+                    # Identify groups that are saturated in the third group
+                    mask &= np.where(gdq[ints, 2, :, :] & saturated, True, False)
+
+                    # Flag the 2nd group for the pixels passing that gauntlet
+                    gdq[ints, 1][mask] |= saturated
```

Comment:
I think we need to use a temporary array here; if we set SATURATED for things passing the gauntlet above in the gdq array, and then grow the gdq array, then we'll be growing many pixels that we didn't intend to which didn't pass the gauntlet.  I.e., perhaps define a zero-valued gdq_temp, set SATURATED for things that pass the gauntlet, grow gdq_temp, then add gdq_temp into gdq[ints,1, ...]